### PR TITLE
[release/v2.14] ship image-loader as new archive for each GitHub release (#6092)

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
What it says on the tin. Includes the bugfix from #6095.

**Does this PR introduce a user-facing change?**:
```release-note
Ship image-loader as new archive for each GitHub release
```

### CODE REVIEW

File: api/hack/ci/ci-github-release.sh [hunk 1]
```diff
@@ -22,119 +22,190 @@ set -euo pipefail
...
   res=$(github_cli \
-    "https://uploads.github.com/repos/$repo/releases/$releaseID/assets?name=${file}" \
-    -H "Accept: application/json" \
-    -H 'Content-Type: application/gzip' \
-    -s --data-binary "@${file}")
-  if echo "${res}" | jq -e '.'; then
-    # it the response contain errors
-    if echo "${res}" | jq -e '.errors[0]'; then
-      for err in $(echo "${res}" | jq -r '.errors[0].code'); do
+    "https://uploads.github.com/repos/$GIT_REPO/releases/$releaseID/assets?name=$(basename "$file")" \
```
Comment (kron4eg) [inline api/hack/ci/ci-github-release.sh:87]: FYI:

```
$ gh release upload -h
Upload asset files to a GitHub Release.

To define a display label for an asset, append text starting with '#' after the
file name.


USAGE
  gh release upload <tag> <files>... [flags]

FLAGS
  --clobber   Overwrite existing assets of the same name

INHERITED FLAGS
      --help              Show help for command
  -R, --repo OWNER/REPO   Select another repository using the OWNER/REPO format

LEARN MORE
  Use 'gh <command> <subcommand> --help' for more information about a command.
  Read the manual at https://cli.github.com/manual

```

Comment (xrstf) [reply api/hack/ci/ci-github-release.sh:87]: Yeah but we don't have it installed in our Docker images. And just for wrapping a single curl call, I don't see much value in adding the gh binary to our Docker images. ^^

**Pull request merged**